src/charm.py

```diff
@@ -159,11 +171,14 @@ def log_level(self):
     @property
     def _default_config(self):
         """Default configuration for the Prometheus workload."""
+        base_path = urlparse(self.external_url).path
+        metrics_path = base_path.rstrip("/") + "/metrics"
```

Comment:
This is almost identical to what's in `self_scraping_job` but structured slightly differently. If there a reason not to just use one code variant and make it a property?

Reply:
Good catch! Fixed in next commit.